cip-0010.md

```diff
@@ -0,0 +1,131 @@
+
+ * **LUT**: A lookup table. Elements are expressed as indexes into this table with the minimum amount of bits required for the total amount of elements in it.
+ * **RLE**: Run-length Encoding, a compression technique used to express multiple values grouped by type. A run-length specifies a group value and amount of elements contained in that group.
+ * **INT**: A big endian 8 byte integer,
+
+# Overview
+
+The main idea is to create a LUT for addreses so they can be expressed by simple indexes, and then compress each amount sent with simple RLE. This would save space by reducing the amount of bits dedicated to expressing the
+addresses. In the most extreme cases, reduction of bits needed to express each address is 168 times than
+standard sends.
+
+# Specification
+
+The message format is constructed in the following way (Where N is the number of recipients, M the number of assets, X the Log base 2 of N rounded up to the nearest integer):
+
+LUT size|LUT address 0|...|LUT address N|Asset Signaling bit 1|Asset SubList 0|...|Asset Signaling Bit 1|Asset Sublist M|Asset Signaling Bit 0|
+--------|-------------|---|-------------|---------------------|---------------|---|---------------------|---------------|---------------------|
+ 16 Bits|   21 bytes  |...|   21 bytes  |         1-bit       |   AssetList   |...|        1-bit        |   AssetList   |        1-bit        |
+
+`AssetList` is built in the following way:
+
+Asset Name|# Z of recipients|LUT Idx 0|Amount Idx 0|...|LUT Idx Z|Amount Idx Z|
+----------|-----------------|---------|------------|---|---------|------------|
+   INT    |     X bits      |  X bits |     INT    |...|  X bits |     INT    |
+
+Note that addresses are only 21 bytes. That's the result of decoding the base58 addresses into binary representation and stripping the last 4 bytes which are just a checksum.
+
+Due to the high bandwidth of transactions created by MPMA sends, and to prevent spam from a given address to DDoS the counterparty network, each address can only transmit an MPMA send once per block. After and address transmits an MPMA send on a given block, subsequent MPMA sends from that same address will be ignored until the next block. This should also be reported as an error from the counterparty-lib client and RPC interfaces.
+
+The MPMA sends messages should by identified by the message type 3 inside a standard, CIP11 compliant, message envelope.
+
+#### Example
+
+The following example is broken into several newlines, but the data stream is continuous on the encoded TX data. Numbers are preceded by a marker, 'x' for hexadecimal numbers, 'b' for bits.
+
+ * `x2000` LUT Size: 2 addresses, 1 bit to represent each one
+ * `x00F3A6B6E4A093E5A5B9DA76977A5270FD4D62553E` Base58 decoded address (into hex, 21 bytes), LUT index 0
+ * `x009BDE9D244E7ABD061B0DCC1DFA8C0D880CD7DA21` Base58 decoded address (into hex, 21 bytes), LUT index 1
+ * `b1` Bit, Valid asset after this
+ * `x0100000000000000` 64 bits XCP asset ID
```

Comment:
Likewise, isn't this `0x0000000000000001`?

Reply:
See above

Reply:
Same as above